src/OrleansSQLUtils/Storage/RelationalOrleansQueries.cs

```diff
@@ -382,886 +255,185 @@ public static Task<RelationalOrleansQueries> CreateInstance(string invariantName
-            }, (selector, resultSetCount, cancellationToken) =>
-            {
-                return Task.FromResult(Convert.ToBase64String(selector.GetValue<byte[]>("ETag")));
-            }).ConfigureAwait(continueOnCapturedContext: false);
-
-            return ret != null ? ret.FirstOrDefault() : null;
+            return ReadAsync(dbStoredQueries.UpsertReminderRowKey, DbStoredQueries.Converters.GetVersion, command =>
+                new DbStoredQueries.Columns(command)
+                {
+                    ServiceId = serviceId,
+                    GrainHash = grainRef.GetUniformHashCode(),
+                    GrainId = grainRef.ToKeyString(),
+                    ReminderName = reminderName,
+                    StartTime = startTime,
+                    Period = period
+                }, ret => ret.First().ToString());
         }
 
-
         /// <summary>
-        /// 
+        /// Deletes a reminder
         /// </summary>
         /// <param name="serviceId">Service ID.</param>
         /// <param name="grainRef"></param>
         /// <param name="reminderName"></param>
         /// <param name="etag"></param>
         /// <returns></returns>
-        internal async Task<bool> DeleteReminderRowAsync(string serviceId, GrainReference grainRef, string reminderName, string etag)
+        internal Task<bool> DeleteReminderRowAsync(string serviceId, GrainReference grainRef, string reminderName,
+            string etag)
         {
-            var ret = await storage.ReadAsync(orleansQueries.DeleteReminderRowKey, command =>
-            {
-                var serviceIdParameter = CreateServiceIdParameter(command, serviceId);
-                command.Parameters.Add(serviceIdParameter);
-
-                var grainIdParameter = CreateGrainIdParameter(command, grainRef.ToKeyString());
-                command.Parameters.Add(grainIdParameter);
-
-                var reminderNameParameter = CreateReminderName(command, reminderName);
-                command.Parameters.Add(reminderNameParameter);
-
-                var etagParameter = CreateEtagParameter(command, etag);
-                command.Parameters.Add(etagParameter);
-            }, (selector, resultSetCount, cancellationToken) =>
-            {
-                return Task.FromResult(selector.GetBoolean(0));
-            }).ConfigureAwait(continueOnCapturedContext: false);
-
-            return ret.First();
+            return ReadAsync(dbStoredQueries.DeleteReminderRowKey, DbStoredQueries.Converters.GetSingleBooleanValue, command =>
+                new DbStoredQueries.Columns(command)
+                {
+                    ServiceId = serviceId,
+                    GrainId = grainRef.ToKeyString(),
+                    ReminderName = reminderName,
+                    Version = etag
+                }, ret => ret.First());
         }
 
-
         /// <summary>
         /// deletes all reminders rows of a service id
```

Comment:
Nit: starts with a small letter, should probably be capitalized and perhaps a point added.